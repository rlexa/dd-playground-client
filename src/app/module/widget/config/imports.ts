import { CommonModule } from '@angular/common';
import { MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';

export const imports = [CommonModule, RouterModule, MatCardModule, MatListModule];
