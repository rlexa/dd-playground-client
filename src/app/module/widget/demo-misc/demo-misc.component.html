<div ddFlexbox="vyt">
  <mat-card class="childv">
    <mat-card-title>Polynomial {{ '(' + transitionalPolynomPoint.join(', ') + ')' }}</mat-card-title>
    <mat-card-content>
      <div ddFlexbox="vys">
        <app-diagram-polynom ddFlexitem="at" [points]="samplePolynomPoints$ | async" [polynomials]="[[1], [1, 1], [1, 1, 1], [1, 1, 1, 1]]"
          (hoveredPoint)="transitionalPolynomPoint = $event"></app-diagram-polynom>
        <div ddFlexbox="hyc">
          <button mat-raised-button color="accent" (click)="reSamplePolynomPoints()">POINTS</button>
        </div>
      </div>
    </mat-card-content>
  </mat-card>

  <mat-card class="childv">
    <mat-card-title>Graphsky</mat-card-title>
    <mat-card-content>
      <div ddFlexbox="hysw">

        <div ddFlexbox="vys">
          <div ddFlexbox="hyc">
            <button mat-button color="accent" (click)="graphskyAddTestData()">+ TEST</button>
            <button mat-button color="warn" (click)="graphskyDrop()">DROP</button>
          </div>
          <app-simple-view class="hovered-elevated" subheader="Stats" [data]="graphskyLog$ | async" [isWidthConstrained]="false" [isDense]="true"></app-simple-view>
        </div>

        <app-simple-view class="hovered-elevated" subheader="All" [data]="graphskyAll$ | async" [isWidthConstrained]="false" [isDense]="true"></app-simple-view>
        <app-simple-view class="hovered-elevated" subheader="Likes" [data]="graphskyLiked$ | async" [isWidthConstrained]="false" [isDense]="true"></app-simple-view>
        <app-simple-view class="hovered-elevated" subheader="Diversity" [data]="graphskyDiversity$ | async" [isWidthConstrained]="false"
          [isDense]="true"></app-simple-view>

      </div>
    </mat-card-content>
  </mat-card>
</div>