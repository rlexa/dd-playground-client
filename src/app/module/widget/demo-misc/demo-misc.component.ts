import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
import { GraphskyService } from 'app/module/service/graphsky-api';
import { arrayFrom } from 'app/util';
import { RxCleanup } from 'dd-rxjs';
import { BehaviorSubject } from 'rxjs';
import { map } from 'rxjs/operators';

const createSamplePoints = (range = 100) => arrayFrom(10 * 2).map(ii => Math.random() * range - range / 2)

@Component({
  selector: 'app-demo-misc',
  templateUrl: './demo-misc.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
  providers: [GraphskyService],
})
export class DemoMiscComponent implements OnDestroy {
  constructor(private readonly graphsky: GraphskyService) { }

  private readonly samplePolynomPointsRange = 100;
  @RxCleanup() readonly samplePolynomPoints$ = new BehaviorSubject(createSamplePoints(this.samplePolynomPointsRange));
  transitionalPolynomPoint = [0, 0];

  readonly graphskyLog$ = this.graphsky.log$;
  readonly graphskyNodes$ = this.graphsky.nodeCount$;
  readonly graphskyAll$ = this.graphsky.change$.pipe(map(() => this.graphsky.query(
    (nodes, _) => nodes
      .filter(node => 'name' in node.data)
      .sort((aa, bb) => aa.data['name'].toString().localeCompare(bb.data['name'].toString()))
      .reduce((acc, node) => (
        {
          ...acc,
          [node.data['name'].toString() + ' ' + node.data['sex'].toString()]: node.out.map(link => link.data['relationship'] + ' ' + link.to.data['name']).join(', ')
        }), {})
  )));
  readonly graphskyLiked$ = this.graphsky.change$.pipe(map(() => this.graphsky.query(
    (_, links) => links
      .filter(link => link.data['relationship'] === 'likes')
      .map(link => link.to)
      .sort((aa, bb) => aa.data['name'].toString().localeCompare(bb.data['name'].toString()))
      .reduce((acc, node) => (
        {
          ...acc,
          [node.data['name'].toString()]: (acc[node.data['name'].toString()] || 0) + 1
        }), {})
  )));
  readonly graphskyDiversity$ = this.graphsky.change$.pipe(map(() => this.graphsky.query(
    (nodes, _) => nodes
      .filter(node => 'sex' in node.data)
      .reduce((acc, node) => (
        {
          ...acc,
          [node.data['sex'].toString()]: (acc[node.data['sex'].toString()] || 0) + 1
        }), {})
  )));

  ngOnDestroy() { }

  reSamplePolynomPoints = () => this.samplePolynomPoints$.next(createSamplePoints(this.samplePolynomPointsRange));

  graphskyAddTestData = () => {
    this.graphsky.add([{ name: 'Alice', sex: 'f' }, { name: 'Bob', sex: 'm' }, { name: 'Clark', sex: 'm' }, { name: 'Donnie', sex: 'm' }]);
    this.graphsky.link([
      { fromNotExact: true, toNotExact: true, data: { relationship: 'likes' }, from: { name: 'Alice' }, to: { name: 'Bob' } },
      { fromNotExact: true, toNotExact: true, data: { relationship: 'likes' }, from: { name: 'Alice' }, to: { name: 'Clark' } },
      { fromNotExact: true, toNotExact: true, data: { relationship: 'likes' }, from: { name: 'Bob' }, to: { name: 'Clark' } },
      { fromNotExact: true, toNotExact: true, data: { relationship: 'likes' }, from: { name: 'Clark' }, to: { name: 'Donnie' } },
      { fromNotExact: true, toNotExact: true, data: { relationship: 'likes' }, from: { name: 'Donnie' }, to: { name: 'Alice' } },
    ]);
  }

  graphskyDrop = () => this.graphsky.drop();
}
