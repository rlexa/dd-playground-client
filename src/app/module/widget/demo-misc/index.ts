export * from './demo-misc.component';
export * from './demo-misc.module';
