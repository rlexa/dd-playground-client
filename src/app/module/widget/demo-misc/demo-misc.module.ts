import {CommonModule} from '@angular/common';
import {NgModule} from '@angular/core';
import {MatButtonModule} from '@angular/material/button';
import {MatCardModule} from '@angular/material/card';
import {RouterModule, Routes} from '@angular/router';
import {ROUTE_ROOT, ROUTE_WILDCARD} from 'src/app/routing';
import {FlexboxModule} from '../../directive/flexbox';
import {SimpleViewModule} from '../simple-view';
import {DemoMiscComponent} from './demo-misc.component';

const ROUTING: Routes = [
  {path: ROUTE_ROOT, component: DemoMiscComponent, pathMatch: 'full'},
  {path: ROUTE_WILDCARD, redirectTo: ROUTE_ROOT},
];

@NgModule({
  imports: [CommonModule, MatButtonModule, MatCardModule, FlexboxModule, SimpleViewModule, RouterModule.forChild(ROUTING)],
  exports: [DemoMiscComponent],
  declarations: [DemoMiscComponent],
})
class DemoMiscModule {}

export {DemoMiscModule, DemoMiscComponent};
