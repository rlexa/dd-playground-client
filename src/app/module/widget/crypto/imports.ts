import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import {
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
} from '@angular/material';
import { RouterModule } from '@angular/router';
import { CryptoApiModule } from 'app/module/service/crypto-api';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  FormsModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
  FlexboxModule,
  CryptoApiModule,
  SimpleViewModule,
];
