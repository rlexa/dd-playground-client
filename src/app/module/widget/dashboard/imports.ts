import { CommonModule } from '@angular/common';
import { MatButtonModule, MatToolbarModule, MatTooltipModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { IconPipeModule } from 'app/module/pipe/icon';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { VersionModule } from 'app/module/widget/version';
import { WidgetsModule } from 'app/widgets/widgets.module';
import { FlexboxModule } from 'dd-flexbox';
import 'hammerjs';

export const imports = [
  CommonModule, RouterModule,
  MatButtonModule, MatToolbarModule, MatTooltipModule, FlexboxModule,
  IconPipeModule, VersionModule,
  // remove
  RoutedContentModule, WidgetsModule,
];
