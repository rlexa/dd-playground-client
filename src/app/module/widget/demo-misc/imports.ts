import { CommonModule } from '@angular/common';
import { MatButtonModule, MatCardModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { GraphskyApiModule } from 'app/module/service/graphsky-api';
import { DiagramPolynomModule } from 'app/module/widget/diagram-polynom';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  FlexboxModule,
  DiagramPolynomModule,
  GraphskyApiModule,
  SimpleViewModule,
];
