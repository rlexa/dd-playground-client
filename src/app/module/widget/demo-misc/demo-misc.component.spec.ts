import { async, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { DemoMiscComponent } from './demo-misc.component';
import { imports } from './imports';

describe('DemoMiscComponent', () => {
  beforeEach(async(() => {
    TestBed.configureCompiler(<any>{ preserveWhitespaces: false }).configureTestingModule({
      imports: [NoopAnimationsModule, RouterTestingModule, ...imports], declarations: [DemoMiscComponent], providers: [],
    }).compileComponents();
  }));

  test('is created', () => {
    const fixture = TestBed.createComponent(DemoMiscComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
