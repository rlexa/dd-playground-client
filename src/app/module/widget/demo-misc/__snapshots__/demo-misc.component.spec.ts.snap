// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoMiscComponent is created 1`] = `
<app-demo-misc
  graphsky={[Function GraphskyService]}
  graphskyAddTestData={[Function Function]}
  graphskyAll$={[Function Observable]}
  graphskyDiversity$={[Function Observable]}
  graphskyDrop={[Function Function]}
  graphskyLiked$={[Function Observable]}
  graphskyLog$={[Function BehaviorSubject]}
  graphskyNodes$={[Function BehaviorSubject]}
  reSamplePolynomPoints={[Function Function]}
  samplePolynomPoints$={[Function BehaviorSubject]}
  samplePolynomPointsRange={[Function Number]}
  transitionalPolynomPoint={[Function Array]}
>
  <div
    ddflexbox="vyt"
    ng-reflect-dd-flexbox="vyt"
    style="display: flex; flex-direction: column; justify-content: flex-start; align-items: stretch; align-content: stretch; flex-wrap: no-wrap;"
  >
    <mat-card
      class="childv mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        Polynomial (0, 0)
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <div
          ddflexbox="vys"
          ng-reflect-dd-flexbox="vys"
          style="display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
        >
          <app-diagram-polynom
            ddflexitem="at"
            ng-reflect-dd-flexitem="at"
            ng-reflect-points="-37.655,-37.655,-37.655,-37.65"
            ng-reflect-polynomials="1,1,1,1,1,1,1,1,1,1"
          >
            <div
              class="svg-container position-relative"
            >
              <svg
                id="svgChart"
                preserveAspectRatio="xMinYMin meet"
                viewBox="0 0 55 375"
              >
                <g
                  transform="translate(35, 5)"
                >
                  <rect
                    class="filler"
                    height="100%"
                    style="fill: transparent;"
                    width="100%"
                  />
                  <g
                    class="content"
                  />
                  <g
                    class="axis-x"
                    transform="translate(0,350)"
                  />
                  <g
                    class="axis-y"
                  />
                </g>
              </svg>
            </div>
          </app-diagram-polynom>
          <div
            ddflexbox="hyc"
            ng-reflect-dd-flexbox="hyc"
            style="display: flex; flex-direction: row; justify-content: flex-start; align-items: center; align-content: stretch; flex-wrap: no-wrap;"
          >
            <button
              class="mat-raised-button mat-accent _mat-animation-noopable"
              color="accent"
              mat-raised-button=""
              ng-reflect-color="accent"
            >
              <span
                class="mat-button-wrapper"
              >
                POINTS
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </div>
        </div>
      </mat-card-content>
    </mat-card>
    <mat-card
      class="childv mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        Graphsky
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <div
          ddflexbox="vys"
          ng-reflect-dd-flexbox="vys"
          style="display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
        >
          <div
            ddflexbox="hycw"
            ng-reflect-dd-flexbox="hycw"
            style="display: flex; flex-direction: row; justify-content: flex-start; align-items: center; align-content: stretch; flex-wrap: wrap;"
          >
            <button
              class="mat-button mat-accent _mat-animation-noopable"
              color="accent"
              mat-button=""
              ng-reflect-color="accent"
            >
              <span
                class="mat-button-wrapper"
              >
                + TEST
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
            <button
              class="mat-button mat-warn _mat-animation-noopable"
              color="warn"
              mat-button=""
              ng-reflect-color="warn"
            >
              <span
                class="mat-button-wrapper"
              >
                DROP
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </div>
          <app-simple-view
            class="ng-tns-c2-0"
            ng-reflect-data="[object Object]"
            ng-reflect-is-dense="true"
            ng-reflect-is-width-constrained="false"
          >
            
            <div
              class="position-relative ng-tns-c2-0 ng-star-inserted"
            >
              <div
                class="scroll-x-auto"
                ddflexbox="hys"
                ng-reflect-dd-flexbox="hys"
                style="max-width: 100%; display: flex; flex-direction: row; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
              >
                
                
                <mat-list
                  class="mat-list-auto-height mat-list ng-tns-c2-0 ng-star-inserted"
                  dense="true"
                >
                  
                  <table
                    class="ng-tns-c2-0 ng-star-inserted"
                  >
                    <tbody
                      class="ng-tns-c2-0"
                    >
                      
                      
                      
                      
                      
                      <tr
                        class="ng-tns-c2-0 ng-star-inserted"
                      >
                        <td
                          class="ng-tns-c2-0"
                          style="vertical-align: top;"
                        >
                          <mat-list-item
                            class="mat-list-item"
                          >
                            <div
                              class="mat-list-item-content"
                            >
                              <div
                                class="mat-list-item-ripple mat-ripple"
                                mat-ripple=""
                                ng-reflect-disabled="true"
                                ng-reflect-trigger="[object HTMLUnknownElement]"
                              />
                              <div
                                class="mat-list-text"
                              />
                              <span
                                class="no-break-text ng-tns-c2-0"
                              >
                                Links
                              </span>
                            </div>
                          </mat-list-item>
                        </td>
                        <td
                          class="ng-tns-c2-0"
                        >
                          <mat-list-item
                            class="mat-list-item ng-trigger ng-trigger-TRIGGER_WobbleX ng-animating"
                            style=""
                          >
                            <div
                              class="mat-list-item-content"
                            >
                              <div
                                class="mat-list-item-ripple mat-ripple"
                                mat-ripple=""
                                ng-reflect-disabled="true"
                                ng-reflect-trigger="[object HTMLUnknownElement]"
                              />
                              <div
                                class="mat-list-text"
                              />
                              <span
                                class="break-text ng-tns-c2-0 value-numeric"
                                ng-reflect-klass="break-text"
                                ng-reflect-ng-class="[object Object]"
                              >
                                
                                
                                
                                
                                
                                
                                
                                <span
                                  class="ng-tns-c2-0 ng-star-inserted"
                                  ng-reflect-ng-class="[object Object]"
                                  style=""
                                >
                                   0 
                                </span>
                              </span>
                            </div>
                          </mat-list-item>
                        </td>
                      </tr>
                      <tr
                        class="ng-tns-c2-0 ng-star-inserted"
                      >
                        <td
                          class="ng-tns-c2-0"
                          style="vertical-align: top;"
                        >
                          <mat-list-item
                            class="mat-list-item"
                          >
                            <div
                              class="mat-list-item-content"
                            >
                              <div
                                class="mat-list-item-ripple mat-ripple"
                                mat-ripple=""
                                ng-reflect-disabled="true"
                                ng-reflect-trigger="[object HTMLUnknownElement]"
                              />
                              <div
                                class="mat-list-text"
                              />
                              <span
                                class="no-break-text ng-tns-c2-0"
                              >
                                Nodes
                              </span>
                            </div>
                          </mat-list-item>
                        </td>
                        <td
                          class="ng-tns-c2-0"
                        >
                          <mat-list-item
                            class="mat-list-item ng-trigger ng-trigger-TRIGGER_WobbleX ng-animating"
                            style=""
                          >
                            <div
                              class="mat-list-item-content"
                            >
                              <div
                                class="mat-list-item-ripple mat-ripple"
                                mat-ripple=""
                                ng-reflect-disabled="true"
                                ng-reflect-trigger="[object HTMLUnknownElement]"
                              />
                              <div
                                class="mat-list-text"
                              />
                              <span
                                class="break-text ng-tns-c2-0 value-numeric"
                                ng-reflect-klass="break-text"
                                ng-reflect-ng-class="[object Object]"
                              >
                                
                                
                                
                                
                                
                                
                                
                                <span
                                  class="ng-tns-c2-0 ng-star-inserted"
                                  ng-reflect-ng-class="[object Object]"
                                  style=""
                                >
                                   0 
                                </span>
                              </span>
                            </div>
                          </mat-list-item>
                        </td>
                      </tr>
                      
                      
                    </tbody>
                  </table>
                </mat-list>
              </div>
            </div>
          </app-simple-view>
          <div
            ddflexbox="hysw"
            ng-reflect-dd-flexbox="hysw"
            style="display: flex; flex-direction: row; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: wrap;"
          >
            <pre
              class="childhv contenthv border"
            >
              All: {}
            </pre>
            <pre
              class="childhv contenthv border"
            >
              Likes: {}
            </pre>
            <pre
              class="childhv contenthv border"
            >
              Diversity: {}
            </pre>
          </div>
        </div>
      </mat-card-content>
    </mat-card>
  </div>
</app-demo-misc>
`;
